dia/tree-parser.rkt

```diff
@@ -113,6 +123,15 @@ on format.
      (cons (if name2 (list name1 name2) name1)
            attribution)]))
 
+(define (leaf->attribution2 x)
```

Comment:
See eed4c313b8f:
```
    create an module lang for appraisal trees

    This "2" language is unstable, since several other improvements are
    planned (mostly to the tree-parsing stage, however). I envision making
    all the changes work in the experiment, stabilising it, and then
    converting existing repos to the new langs.
```

and ffdb729ba4d, which notes the duplicated code needing refactored.

Paging this back in, I think the idea here was that I needed a new `leaf->attribution` for the new (uniform contributor) syntax. It's only used in `#lang dia/attribution2` via `read-attribution-tree2` for the moment, but eventually we'll migrate as mentioned above.